async-nats/tests/client_tests.rs

```diff
@@ -562,4 +563,14 @@ mod client {
             .unwrap();
         inbox_wildcard_subscription.next().await.unwrap();
     }
+
+    #[tokio::test]
+    async fn connection_state() {
+        let server = nats_server::run_basic_server();
+        let client = async_nats::connect(server.client_url()).await.unwrap();
+        assert_eq!(State::Connected, client.connection_state());
+        drop(server);
+        tokio::time::sleep(Duration::from_secs(1)).await;
+        assert_eq!(State::Disconnected, client.connection_state());
```

Comment:
Test for reconnect too?

Reply:
hard to do here. Added reconnect check in reconnection general test.